src/test/signin.test.js

```diff
@@ -0,0 +1,244 @@
+import bcrypt from 'bcrypt';
+
+import models from '../db/models';
+import app from '../index';
+
+chai.use(chaiHttp);
+
+const { expect } = chai;
+const signinUrl = '/api/v1/auth/signin';
+const { User } = models;
+
+const testUserDetails = {
+  first_name: 'Olawumi',
+  last_name: 'Yusuff',
+  email: 'user@test.com',
+  password: '12345678'
+};
+let testUser;
+
+
+before((done) => {
+  bcrypt.hash(testUserDetails.password, 10)
+    .then((hash) => User.create({
+      first_name: testUserDetails.first_name,
+      last_name: testUserDetails.last_name,
+      email: testUserDetails.email,
+      password: hash
+    }))
+    .then((user) => {
+      testUser = user;
+      done();
+    })
+    .catch((e) => done(e));
+});
+
+after((done) => {
+  User.destroy({
+    where: {
+      email: testUserDetails.email
+    }
+  })
+    .then(() => done())
+    .catch((e) => done(e));
+});
+
+describe(`POST ${signinUrl}`, () => {
+  describe('SUCCESS', () => {
+    it('should sign in a user successfully', async () => {
+      const res = await chai.request(app)
+        .post(signinUrl)
+        .send({
+          email: testUserDetails.email,
+          password: testUserDetails.password
+        });
+
+      expect(res.status).to.equal(200);
+      expect(res.body).to.be.an('object');
+      expect(res.body).to.have.property('status');
+      expect(res.body.status).to.equal('success');
+      expect(res.body).to.have.property('data');
+      expect(res.body.data).to.be.an('object');
+      expect(res.body.data).to.have.property('role');
+      expect(res.body.data).to.have.property('isVerified');
+      expect(res.body.data).to.have.property('id');
+      expect(res.body.data).to.have.property('first_name');
+      expect(res.body.data).to.have.property('last_name');
+      expect(res.body.data).to.have.property('email');
+      expect(res.body.data).to.not.have.property('password');
+      expect(res.body.data).to.have.property('updatedAt');
+      expect(res.body.data).to.have.property('createdAt');
+      expect(res.body.data).to.have.property('gender');
+      expect(res.body.data).to.have.property('birth_date');
+      expect(res.body.data).to.have.property('preferred_language');
+      expect(res.body.data).to.have.property('preferred_currency');
+      expect(res.body.data).to.have.property('location');
+      expect(res.body.data).to.have.property('token');
+      expect(res.body.data.token).to.be.a('string');
+      expect(res.body.data.role).to.equal(testUser.role);
+      expect(res.body.data.isVerified).to.equal(testUser.isVerified);
+      expect(res.body.data.id).to.equal(testUser.id);
+      expect(res.body.data.first_name).to.equal(testUser.first_name);
+      expect(res.body.data.last_name).to.equal(testUser.last_name);
+      expect(res.body.data.email).to.equal(testUser.email);
+      expect(res.body.data.gender).to.equal(testUser.gender);
+      expect(res.body.data.preferred_language).to.equal(testUser.preferred_language);
```

Comment:
check indentation

Reply:
OK, will do. Though I believe ESLint would have flagged it.